packages/cli/internal/pkg/cli/workflow/manager.go

```diff
@@ -348,6 +361,37 @@ func (m *Manager) parseInputToArguments() {
 	m.arguments = []string{arguments}
 }
 
+// writeTempManifest writes the inputsFile included in the command line to the temporary MANIFEST.json located in temp directory
+// This function is only called if there is a path included in the command line with the --inputsFile flag
+func (m *Manager) writeTempManifest() {
+	if m.err != nil || m.inputsPath == "" {
+		return
+	}
+	m.manifestPath = filepath.Join("temp", "MANIFEST.json")
+	log.Debug().Msgf("Reading %s", m.manifestPath)
+	bytes, err := m.Storage.ReadAsBytes(m.manifestPath)
+	if err != nil {
+		m.err = err
+		return
+	}
+	var data ManifestProps
+	if err := json.Unmarshal(bytes, &data); err != nil {
+		m.err = err
+		return
+	}
+	data.InputFileURLs = append(data.InputFileURLs, m.inputsPath)
+	bytes, err = json.Marshal(data)
+	if err != nil {
+		m.err = err
+		return
+	}
+	err = m.Storage.WriteFromBytes(m.manifestPath, bytes)
+	if err != nil {
+		m.err = err
+		return
+	}
+}
+
 func (m *Manager) uploadInputsToS3() {
```

Comment:
qq: have you update the logic to upload the updated temp `MANIFEST.json` into s3 bucket and remove unnecessary logic to upload files from `--inputsFile`?